File: backend/src/models/Reply.ts
import mongoose from 'mongoose';

const ReplySchema = new mongoose.Schema({
    thread: { type: mongoose.Schema.Types.ObjectId, ref: 'Thread', required: true },
    content: { type: String, required: true },
    author: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
    likeCount: { type: Number, default: 0 },
    likes: [{ type: mongoose.Schema.Types.ObjectId, ref: 'User' }],
    isEdited: { type: Boolean, default: false },
    createdAt: { type: Date, default: Date.now },
    updatedAt: { type: Date, default: Date.now }
});

export const Reply = mongoose.model('Reply', ReplySchema);

File: mobile/app/src/main/java/com/example/derdine/data/model/Reply.kt
package com.example.derdinesokayim.data.model

data class Reply(
    val id: String,
    val threadId: String,
    val content: String,
    val author: User,
    val createdAt: Long,
    val likeCount: Int = 0,
    val isLiked: Boolean = false,
    val isEdited: Boolean = false
)
